.github/workflows/ocp-test-profiles.yaml

```diff
@@ -0,0 +1,143 @@
+          path: ctf
+      # https://github.com/actions/checkout/issues/766
+      - name: Set git safe directory
+        run: git config --global --add safe.directory "$GITHUB_WORKSPACE"
+      - name: Find forking point
+        env:
+          BASE_BRANCH: ${{ github.base_ref }}
+        run: echo "FORK_POINT=$(git merge-base origin/$BASE_BRANCH ${{ github.event.pull_request.head.sha }})" >> $GITHUB_OUTPUT
+        id: fork_point
+      - name: Detect content changes in the PR
+        run: python3 ./ctf/content_test_filtering.py pr --base ${{ steps.fork_point.outputs.FORK_POINT }} --remote_repo ${{ github.server_url }}/${{ github.repository }}  --verbose --rule --output json ${{ github.event.pull_request.number }} > ctf-output.json
+      - name: Test if there are no content changes
+        run: echo "CTF_OUTPUT_SIZE=$(stat --printf="%s" ctf-output.json)" >> $GITHUB_OUTPUT
+        id: ctf
+      - uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        with:
+          name: ctf-output
+          path: ctf-output.json
+      - name: Print changes to content detected if any
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        run: cat ctf-output.json
+      - name: Get product attribute
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' }}
+        id: product
+        uses: notiz-dev/github-action-json-property@a5a9c668b16513c737c3e1f8956772c99c73f6e8 # v0.2.0
+        with:
+          path: 'ctf-output.json'
+          prop_path: 'product'
+
+      - name: Build product OCP and RHCOS content
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' && (contains(steps.product.outputs.prop, 'ocp4') || contains(steps.product.outputs.prop, 'rhcos4')) }}
+        run: ./build_product -d ocp4 rhcos4
+
+      - name: Process list of rules into a list of product-profiles to test
+        if: ${{ steps.ctf.outputs.CTF_OUTPUT_SIZE != '0' && (contains(steps.product.outputs.prop, 'ocp4') || contains(steps.product.outputs.prop, 'rhcos4')) }}
+        id: profiles_to_test 
+        run: |
+          OCP_VERSIONS=(4.17 4.16)
+          RULES=$(cat ctf-output.json | jq -r '.rules[]')
+
+          # Let's grab one profile for each changed rule
+          PROFILES=()
+          ALL_PROFILES=()
+
+          # Let's consistently grab a random profile for each rule, in order to do that we use the
+          # PR number as the seed
+          RANDOM=${{ github.event.pull_request.number }}
+          for rule in $RULES; do
+            readarray -t TEMP <<< $(grep -lr -e "- ${rule}\$" build/*/profiles | sort)
+
+            # Let's ilter out profiles for which we don't have a CI job configured
+            # Here is an example of how to quicly update this variable in the future
+            # TESTED_PROFILES=$(grep -r PROFILE= ./ComplianceAsCode-content-master__4.16.yaml | sort -u | sed 's/.*export PROFILE=\(.*\)/\1/')
```

Comment:
This must be run from the openshift/releases repository, right?

Reply:
I don't have a good work around for this.
We could clone the repo and parse the files, or issue a `/test` comment and parse the comment from the ci bot.

I added it because we don't have tests for all profile ids.
You can see in previous comments that `pci-dss-3-2` was chosen for tests, but we don't have CI wired up for that.

Reply:
I have changed that to actually curl the job config file and grep it.
So unless the job format  or the test command changes, it should keep it self in sync with the tested profiles.